src/node/blockstorage.cpp

```diff
@@ -397,6 +397,15 @@ bool IsBlockPruned(const CBlockIndex* pblockindex)
     return (fHavePruned && !(pblockindex->nStatus & BLOCK_HAVE_DATA) && pblockindex->nTx > 0);
 }
 
+const CBlockIndex* GetFirstStoredBlock(const CBlockIndex* start_block) {
+    AssertLockHeld(::cs_main);
+    const CBlockIndex* last_block = start_block;
+    while (last_block->pprev && (last_block->pprev->nStatus & BLOCK_HAVE_DATA)) {
```

Comment:
In commit "refactor: Introduce GetFirstStoredBlock helper function" (8323cce2aa76123bc385c7927701d42282b0bd26)

Previous version of this function in 19610999595ec557fc602c8cefb930c2dfc1ad67 had a `CHECK_NONFATAL(start_block)` line, which I think was good because it prevented UB if a null start block was pass, which seems like an easy mistake to make. IMO it would be nice to restore this, or add `assert(start_block)` since callers are now using CHECK_NONFATAL.

Reply:
I think the `assert` makes sense here now, added.